### ISSUE TITLE
Q: Start procedure for tests that do not begin with a door opening

### ISSUE THREAD

**Issue by RemiFabre** [RoboCupAtHome/RuleBook#578]
The "3.7.8 Start signal" section doesn't cover the default start procedure when the test doesn't start by a door opening.

For example, in the test _Receptionist_ the robot should start by going towards the door and try opening it.

A simple solution would be that the team is responsible for providing a one-button start procedure so that the referee can start the test when everyone is ready. In most cases this would be pressing enter on the keyboard. Is this ok?

**Comment by moriarty** [comment]
Yes, but without Keyboard.

The robot should communicate the start procedure. When the robot is ready to start (if using ROS- all nodes are up and ready) then it should indicate it is ready, and give some simple “one touch” instruction to the referee. The referee will do as they were told, or at least try to do what they understood.

Most teams using the Toyota HSR in DSPL, use a press of the wrist joint.

Other teams, use some speech command... but be careful because it’s noisy and some keywords are spoken frequently just before the test starts.

We always recommend a backup, such as an aruco marker.

The problem with Keyboards are that it’s too hard to control and we want to ensure that the robots are autonomous, if we see teams touching a Keyboard at the Externally Compute Resource Area (ECRA)- we will yell at them and potentially end their run.

**Comment by moriarty** [comment]
Teleoping the robot, using a keyboard to press enter, and then pulling out the USB cable, should be the the “plan z”... or not allowed at all...
Although it has been allowed in the past... I think we should strongly discourage it.

I think we’d accept a PR which clarifies this rule.

**Comment by kyordhel** [comment]
# TL;DR: Pressing Enter is [borderline] OK.
1. **Touching the robot afterwards causes immediate disqualification**.
    - This includes pulling/disconnecting a keyboard
2. Moving mouse and click are two actions, therefore not allowed.
3. The use of **wireless keyboards**/mouses counts as remote controlling, hence **disqualification**.
    - Remote computers can't be touched. Consequently, starting the robot from ECRA is remote controlling, i.e. forbidden.
4. The use of alternate HRI methods is highly encouraged.

## Explanation
I must say the rules seem quite clear to me
![image](https://user-images.githubusercontent.com/8583949/53954816-93016800-40d7-11e9-860d-79a92601ae15.png)

My explanation derives from §3.7.8, §3.7.5, and §3.5.1 respectively. You are allowed to use any button to start the robot, that would include a single keyboard strike, however teleop and touching are strictly forbidden, hence you can only use a keyboard attached to the robot. Likewise, people in ECRA is not allowed to interact with the computers there, so starting the robot from ECRA is also not allowed.

Being said that, thanks @RemiFabre for pointing out this issue. I must say I agree with @moriarty in that we should disallow using keyboards or any non-intuitive user interface to start a robot.

**Comment by RemiFabre** [comment]
Thank you guys for the answers. When referring to the keyboard I was thinking about the laptop's keyboard that most robots have. The robot would shout something like "Please press the Enter key on my back when you are ready to start the test".

We'll probably implement a wrist twist (but not all robots have the sensors to do so) or a voice command if we're confident enough :)

**Comment by johaq** [comment]
There are tons of viable options so keyboards should not be necessary.
Some robots have bumpers or you can block sensors. I believe TechU blocks their laser scan with a little box and lifting it is the start signal etc.

**Comment by kyordhel** [comment]
Why not a Vizbox start signal? Or perhaps all robots should start outside. 

I agree, no more Kbd.

**Comment by RemiFabre** [comment]
A Vizbox start signal would be nice indeed. As stated in #577 , we'll be working on using Vizbox. Ideally, the robot would have a small touch screen on his head that displays the Vizbox and offers a few interactions. Not sure that this will be ready for this year though.

If blocking the laser scan with a box is OK, then that will be our default behavior since it's the same than waiting for a door to open.

**Comment by kyordhel** [comment]
@RemiFabre  can we consider this as solved?

**Comment by RemiFabre** [comment]
I saw the pull request, while it clarifies what the "plan B" is for the starting procedures, it doesn't clarify what the default expected behavior is for tests that do not start by a door opening. I would suggest blocking the robot path with a barrier/box because : 
- It's the same behavior than the door opening
- It's reliable
- It is clear for the public what caused the test to start

What about adding that information in each test, in the "Start location" section? As in:

> Start location: The robot starts outside the arena. When the door opens, the robot moves towards the Instruction Point.

**Comment by RemiFabre** [comment]
(I can PR this if needed)

**Comment by johaq** [comment]
Tests that do not start with a door opening, start with the referee telling a team member to start the robot. That team member is then expected to start the robot according to the rules (no typing etc.)

**Comment by kyordhel** [comment]
I can't but disagree with @RemiFabre's plan B. I simply wouldn't know which sensor to block, and this is not intuitive by any means to make it a standard. It works good for TU/e but doesn't mean it will work for others.

There is an award for alternate HRI interfaces that can have a start button. There is the Vizbox initiative, as well as clear-enough specs on how a physical start-button should be. On top of that, it is OK, to use QR codes and voice commands (which btw are given by the team).

Any other alternate method must be approved by the TC during robot inspection.

**Comment by RemiFabre** [comment]
Fair enough, thank you for the answer. I'm closing this.
